opentelemetry-sdk/src/opentelemetry/sdk/error_handler/__init__.py

```diff
@@ -105,26 +106,26 @@ def __new__(cls) -> "GlobalErrorHandler":
 
         return cls._instance
 
-    def __enter__(self):
+    def __enter__(self) -> None:
         pass
 
     # pylint: disable=no-self-use
-    def __exit__(self, exc_type, exc_value, traceback):
-        if exc_value is None:
+    def __exit__(self, exc_type, exc_value, traceback) -> Optional[bool]:  # type: ignore[no-untyped-def]
```

Comment:
Use type hints mentioned here: https://stackoverflow.com/a/58808179

And remove the `type ignore`.